file_store/src/traits/msg_timestamp.rs

```diff
@@ -0,0 +1,25 @@
+use crate::{error::DecodeError, Result};
+use chrono::{DateTime, TimeZone, Utc};
+
+pub trait MsgTimestamp {
+    fn to_timestamp(self) -> Result<DateTime<Utc>>;
+    fn to_timestamp_millis(self) -> Result<DateTime<Utc>>;
+    fn to_timestamp_nanos(self) -> Result<DateTime<Utc>>;
+}
+
+impl MsgTimestamp for u64 {
+    fn to_timestamp(self) -> Result<DateTime<Utc>> {
```

Comment:
Should these be called `from_timestamp...` since we are converted from a timestamp to a `DateTime<Utc>`??

Reply:
heh.. that depends on what you think the canonical timestamp is.. When I wrote this I thought the meaningful timestamp _is_ the DateTime so you're converting from the wire format _to_ a DateTime